### DISCUSSION TITLE
Combination of pairs / Unique pairs?

### DISCUSSION THREAD

**Discussion by SuperJMN** [morelinq/MoreLINQ#1083] [Q&A]
I would like something like Cartesian, but considering that (a, b) <==> (b, a).

So you would simply create a distance matrix with some code like:

`var distances = cities.UniquePairs(cities, (a, b) => (A: a, B: b, Distance: Random.Shared(200)));`

**Comment by atifaziz** [comment]
I think you might looking for [`Subsets`](https://morelinq.github.io/4.1/ref/api/html/Overload_MoreLinq_MoreEnumerable_Subsets.htm) and specifically the [overload that accepts a size](https://morelinq.github.io/4.1/ref/api/html/M_MoreLinq_MoreEnumerable_Subsets__1_1.htm). If you use a size of 2, you should get your unique pairs.

```c#
using System;
using System.Linq;
using MoreLinq;

var cities = new[]
{
    "New York",
    "Los Angeles",
    "Chicago",
    "Houston",
    "Phoenix"
};

var data =
    from e in cities.Subsets(2).Zip(MoreEnumerable.Random(200))
    select (e.First[0], e.First[1], e.Second);

foreach (var (a, b, distance) in data)
    Console.WriteLine($"({a}, {b}) -> Distance: {distance}");
```

This should produce the output:

    (New York, Los Angeles) -> Distance: 181
    (New York, Chicago) -> Distance: 115
    (New York, Houston) -> Distance: 72
    (New York, Phoenix) -> Distance: 68
    (Los Angeles, Chicago) -> Distance: 108
    (Los Angeles, Houston) -> Distance: 36
    (Los Angeles, Phoenix) -> Distance: 50
    (Chicago, Houston) -> Distance: 19
    (Chicago, Phoenix) -> Distance: 21
    (Houston, Phoenix) -> Distance: 140
